sdktests/common_tests_stream_request.go

```diff
@@ -0,0 +1,162 @@
+			for _, withReasons := range []o.Maybe[bool]{o.None[bool](), o.Some(false), o.Some(true)} {
+				// The reason we use 3 states here instead of 2 is to verify that the SDK uses a default
+				// of false if we *don't* set the property.
+
+				t.Run(fmt.Sprintf("evaluationReasons set to %s", withReasons), func(t *ldtest.T) {
+					for _, method := range c.availableFlagRequestMethods() {
+						t.Run(string(method), func(t *ldtest.T) {
+							dataSource, configurers := c.setupDataSources(t, nil)
+
+							_ = NewSDKClient(t, c.baseSDKConfigurationPlus(
+								append(configurers,
+									WithClientSideConfig(servicedef.SDKConfigClientSideParams{
+										EvaluationReasons: withReasons,
+									}),
+									c.withFlagRequestMethod(method),
+								)...)...)
+
+							request := dataSource.Endpoint().RequireConnection(t, time.Second)
+
+							var queryMatcher m.Matcher
+							if withReasons.Value() {
+								queryMatcher = m.MapOf(
+									m.KV("withReasons", m.Items(m.Equal("true"))),
+								)
+							} else {
+								queryMatcher = m.AnyOf(
+									m.MapOf(
+										m.KV("withReasons", m.Items(m.Equal("false"))),
+									),
+									m.MapOf(),
+								)
+							}
+							m.In(t).For("query string").Assert(request.URL.Query(), queryMatcher)
+						})
+					}
+				})
+			}
+		})
+	}
+}
+
+func (c CommonStreamingTests) RequestUserProperties(t *ldtest.T) {
+	t.RequireCapability(servicedef.CapabilityClientSide) // server-side SDKs do not send user properties in stream requests
+
+	t.Run("user properties", func(t *ldtest.T) {
+		for _, method := range c.availableFlagRequestMethods() {
+			t.Run(string(method), func(t *ldtest.T) {
+				dataSource, configurers := c.setupDataSources(t, nil)
+
+				user := lduser.NewUserBuilder(c.userFactory.NextUniqueUser().GetKey()).
+					Name("a").
+					Email("b").AsPrivateAttribute().
+					Custom("c", ldvalue.String("d")).
+					Build()
+				userJSONMatcher := m.JSONMap().Should(m.MapIncluding(
+					m.KV("key", m.Equal(user.GetKey())),
+					m.KV("name", m.Equal("a")),
+					m.KV("email", m.Equal("b")),
+					m.KV("custom", m.MapOf(
+						m.KV("c", m.Equal("d")),
+					)),
+					m.KV("privateAttributeNames", m.Items(m.Equal("email"))),
+				))
+				_ = NewSDKClient(t, c.baseSDKConfigurationPlus(
+					append(configurers,
+						WithClientSideConfig(servicedef.SDKConfigClientSideParams{
+							InitialUser: user,
+						}),
+						c.withFlagRequestMethod(method),
+					)...)...)
+
+				request := dataSource.Endpoint().RequireConnection(t, time.Second)
+
+				if method == flagRequestREPORT {
+					m.In(t).For("request body").Assert(request.Body, m.AllOf(
+						m.Not(m.BeNil()),
+						userJSONMatcher))
+				} else {
+					m.In(t).For("request body").Assert(request.Body, m.Length().Should(m.Equal(0)))
+
+					pathParts := strings.Split(strings.TrimPrefix(request.URL.Path, "/"), "/")
+					expectedPathComponents := h.IfElse(c.sdkKind == mockld.JSClientSDK, 3, 2)
```

Comment:
This logic regarding the path is a bit clunky - I've improved it in the next PR.